005_team_3_agent_design/005_team_3_agent_design.tex

```diff
@@ -1 +1,194 @@
+    \item When the tower reshuffles the agents (which each agent knows by checking what floor it is on) mood changes depending on the relative position of the new floor to the prior floor.  
+    \item The agent attempts to eat every day. The quantity depends on treaties, the messages it has received and whether other agents have requested it to leave or eat a certain amount of food or a calculation based on the agent’s current values. More information on this can be found in \Cref{sec:food_intake}. 
+    \item Finally, the agent will attempt to communicate with other agents and create a social network via message sending and treaty proposition. More information on this is can be found in \Cref{sec:agent_communication}. 
+  \end{enumerate}
+
+\subsection{Agent Knowledge}
+Agent 3 will be able to make decisions when receiving messages or after signing treaties and remember these until it eats. This information is used to show that a predetermined decision has been taken instead of “impulsively” eating food depending solely on the agent’s hunger and state of mind. The decisions that the agent takes include:
+\begin{enumerate}
+    \item The floors the agent has been in before. This information will affect the agent’s mood after being reshuffled, as it would remember where they have been and know if the new floor is a good or bad one.  
+    \item The last recorded HP value. Used to see changes to the agent's health.
+    \item The agent’s friendship network. It is aware of the people it has met during their time in the tower. The agent’s friends are identified and are considered more or less close to the agent with an affection parameter that ranges from 0 to 1, with 0  representing a complete dislike for the other agent.
+    \item The amount of food last eaten.
+    \item The amount of food last seen on the platform.
+    \item A moving average of food consumed.
+    \item The age of the agent.
+    \item The treaty we have proposed until it is accepted or rejected.
+    \item The estimation of the reshuffle period.
+    \item The HP our neighbour above last told us it had.
+    \item The HP our neighbour below last told us it had.
+\end{enumerate}
+
+\subsection{Agent Generation}
+The defining variables of our agents are randomised when it is generated in order to avoid preconditioned agents and mimic different personalities. All three variables are defined to be in the range 0 to 100. Morality and mood are randomly allocated to a value in the range, but stubbornness is limited to a number between 0 and 75. As explained in our agent parameters, stubbornness represents the effect of external influences on an agent. A stubbornness of 100 would mean this agent is completely unwilling to listen and would ignore all incoming messages, effectively making them an agent that disrupts the tower. In order for this to not be the case, we added the artificial upper bound to the value of stubbornness.\par 
```

Comment:
```suggestion
The defining variables of the agents are randomised when it is generated in order to avoid preconditioned agents and mimic different personalities. All three variables are defined to be in the range 0 to 100. Morality and mood are randomly allocated to a value in the range, but stubbornness is limited to a number between 0 and 75. As explained in our agent parameters, stubbornness represents the effect of external influences on an agent. A stubbornness of 100 would mean this agent is completely unwilling to listen and would ignore all incoming messages, effectively making them an agent that disrupts the tower. In order for this to not be the case, we added the artificial upper bound to the value of stubbornness.\par 
```